File: LeetCode/Greedy/Python/Leetcode670-maximum-swap.py
class Solution:
    def maximumSwap(self, num: int) -> int:
        rec = []
        while num:
            rec.append(num % 10)
            num //= 10
        n = len(rec)
        for i in range(n - 1,-1,-1):
            rmx,idx = rec[i],i
            for j in range(i - 1,-1,-1):
                if rec[j] >= rmx: 
                    rmx = rec[j]
                    idx = j
            if rmx > rec[i]:
                rec[i],rec[idx] = rec[idx],rec[i]
                break
        ans = 0
        for i in range(n - 1,-1,-1):
            ans = ans * 10 + rec[i]
        return ans

File: LeetCode/Greedy/Rust/Leetcode670-maximum-swap.rs
impl Solution {
    pub fn maximum_swap(mut num: i32) -> i32 {
        let mut rec = vec![];
        while num != 0 {
            rec.push(num % 10);
            num /= 10;
        }
        let n = rec.len();
        for i in (0..n).rev() {
            let (mut rmx,mut idx) = (rec[i],i);
            for j in (0..i).rev() {
                if rec[j] >= rmx {
                    rmx = rec[j];
                    idx = j;
                }
            }
            if rmx > rec[i] {
                rec[idx] = rec[i];
                rec[i] = rmx;
                break
            }
        }
        let mut ans = 0;
        for i in (0..n).rev() {
            ans = ans * 10 + rec[i];
        }
        ans 
    }
}
